pennylane/templates/subroutines/controlled_sequence.py

```diff
@@ -0,0 +1,206 @@
+r"""
+Contains the ControlledSequence template.
+"""
+from copy import copy
+import pennylane as qml
+
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+from pennylane.ops.op_math.symbolicop import SymbolicOp
+
+
+class ControlledSequence(SymbolicOp, Operation):
+    r"""Creates a sequence of controlled gates raised to decreasing powers of 2. Can be used as
+    a sub-block in building a `quantum phase estimation <https://en.wikipedia.org/wiki/Quantum_phase_estimation_algorithm>`__
+    circuit.
+
+    Given an :class:`~.Operator` and a list of control wires, this template creates a sequence of
+    controlled gates, one for each control wire, with the base :class:`~.Operator` raised to
+    decreasing powers of 2:
+
+    .. figure:: ../../_static/templates/subroutines/big_ctrl.png
+        :align: center
+        :width: 40%
+        :target: javascript:void(0);
+
+    Args:
+        base (Operator): the phase estimation unitary, specified as an :class:`~.Operator`
+        control (Union[Wires, Sequence[int], or int]): the wires to be used for control
+
+    Raises:
+        ValueError: if the wires in ``control`` and wires on the ``base`` operator share a common
+            element
+
+    .. seealso:: :class:`~.QuantumPhaseEstimation`
+
+    **Example**
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires = 4)
+
+        @qml.qnode(dev)
+        def circuit():
+
+            for i in range(3):
+                qml.Hadamard(wires = i)
+
+            qml.ControlledSequence(qml.RX(0.25, wires = 3), control = [0, 1, 2])
+
+            qml.adjoint(qml.QFT)(wires = range(3))
+
+            return qml.probs(wires = range(3))
+
+    >>> print(circuit())
+    [0.92059345 0.02637178 0.00729619 0.00423258 0.00360545 0.00423258 0.00729619 0.02637178]
+
+    """
+
+    def _flatten(self):
+        return (self.base,), (self.control,)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        return cls(data[0], control=metadata[0])
+
+    def __init__(self, base, control, id=None):
+        control_wires = Wires(control)
+
+        if len(Wires.shared_wires([base.wires, control_wires])) != 0:
+            raise ValueError("The control wires must be different from the base operation wires.")
+
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["base"] = base
+
+        self._name = "ControlledSequence"
+
+        super().__init__(base, id=id)
+
+    @property
+    def hash(self):
+        return hash(
+            (
+                str(self.name),
+                self.control,
+                self.base.hash,
+            )
+        )
+
+    @property
+    def control(self):
+        """The control wires for the sequence"""
+        return self.hyperparameters["control_wires"]
+
+    @property
+    def wires(self):
+        return self.base.wires + self.control
+
+    @property
+    def has_matrix(self):
+        return False
```

Comment:
Don't need this either right? Since the by default this is only true if a `matrix()` method is defined.

Reply:
Edit: We do need this as the base method from SymbolicOp relies on base.